great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -372,7 +377,8 @@ def get_batch_list_from_batch_request(
             batch_spec_passthrough=None,
         )
 
-        batch_metadata = copy.deepcopy(batch_request.options)
+        batch_metadata = copy.deepcopy(self.batch_metadata) or {}
+        batch_metadata.update(copy.deepcopy(batch_request.options))
```

Comment:
I think this is where the config variable substitution will also happen, so I'm planning to revisit this in a subsequent PR.

Reply:
Do we want to put the data asset meta data in a sub-dictionary with some specially defined key? Like this, it is possible that the batch_request options will overwrite the asset metadata.